### ISSUE TITLE
ValueError: Can't specify drop_axis and new_axis together

### ISSUE THREAD

**Issue by cgohlke** [dask/dask#2253]
I am using `map_blocks` with a function that analyzes a 3-dimensional array (a big time series of images)
and returns an array where the first dimension is dropped and two new dimensions are added, e.g. `tyx -> yxab`.

The following example works with dask 0.11.1 but fails with the current version 0.25.2, raising a ValueError:

```Python
import numpy as np
import dask.array as da


def function(image_timeseries, radius, angles, bins):
    """Return new array: drop first dimension, append 2 new dimensions."""
    times, length, width = image_timeseries.shape
    return np.zeros(shape=(length-2*radius, width-2*radius, angles, bins),
                    dtype='float32')


def map_blocks(function, image_timeseries, radius=6, angles=32, bins=32,
               chunk=32):
    """Run function on small overlapping blocks of image timeseries."""
    times, length, width = image_timeseries.shape

    # create dask ghost array of overlapping blocks
    data = da.from_array(image_timeseries, chunks=(times, chunk, chunk))
    data = da.ghost.ghost(data, depth=(0, radius, radius), boundary=(0, 1, 1))

    result = da.core.map_blocks(
        function,
        data,
        dtype='float32',
        chunks=(chunk, chunk, angles, bins),
        drop_axis=0,
        new_axis=(2, 3),
        radius=radius,
        angles=angles,
        bins=bins
        ).compute()

    # trim off the borders
    return result[radius:-radius, radius:-radius]


image_timeseries = np.zeros((2**10, 256, 256), dtype='uint16')
result = map_blocks(function, image_timeseries)
print(result.shape)  # (244, 244, 32, 32)
```

```Python
File "dask_issue.py", line 38, in <module>
  result = map_blocks(function, image_timeseries)
File "dask_issue.py", line 30, in map_blocks
  bins=bins
File "X:\Python36\Lib\site-packages\dask\array\core.py", line 614, in map_blocks
  raise ValueError("Can't specify drop_axis and new_axis together")

builtins.ValueError: Can't specify drop_axis and new_axis together
```

**Comment by mrocklin** [comment]
What is your intention with both dropping axes and adding new ones at the same time?

**Comment by cgohlke** [comment]
> What is your intention with both dropping axes and adding new ones at the same time?

The intention is to make the `map_blocks` function aware that the callable function removes the first dimension and creates two new dimensions at the end wrt the input array.
That is what I understand from the [documentation](http://dask.pydata.org/en/latest/array-api.html#dask.array.core.map_blocks) and it worked as expected with dask 0.11.1. Am I missing something?

**Comment by jcrist** [comment]
Sorry about that - this was removed as it wasn't clear what dropping and adding axis at the same time would do. If we define `drop_axis` as always happening before `new_axis`, then I think we can support this (I see the use here). Fixed in #2264.

**Comment by cgohlke** [comment]
Thank you. The patch #2264 works for me.
